src/checkout.js

```diff
@@ -0,0 +1,150 @@
+// bypasses the Checkout step to ask user for contributions
+import html from "../html/popup.html";
+import { centreElement, addAlbumToCart } from "./utilities.js";
+import Logger from "./logger";
+
+const albumsToPurchase = [1609998585];
+
+export default class Checkout {
+  constructor() {
+    this.log = new Logger();
+
+    this.config;
+    this.dialog;
+    this.checkout_button_sub;
+
+    this.checkoutButtonClicked = Checkout.checkoutButtonClicked.bind(this);
+    this.applyConfig = Checkout.applyConfig.bind(this);
+    this.closeDialogAndGoToCart = Checkout.closeDialogAndGoToCart.bind(this);
+    this.addAlbumToCart = addAlbumToCart.bind(this);
+
+    this.yesButtonClicked = Checkout.yesButtonClicked.bind(this);
+    this.noButtonClicked = Checkout.noButtonClicked.bind(this);
+
+    try {
+      this.port = chrome.runtime.connect(null, { name: "bandcamplabelview" });
+    } catch (e) {
+      if (e.message.includes("Error in invocation of runtime.connect")) {
+        this.log.error(e);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  init() {
+    this.log.info("Loaded Checkout");
+
+    this.checkout_button_sub = Checkout.replaceCheckoutButton();
+    this.checkout_button_sub.addEventListener(
+      "click",
+      this.checkoutButtonClicked
+    );
+
+    this.dialog = Checkout.createDialog();
+
+    this.dialog
+      .querySelector("#yes")
+      .addEventListener("click", this.yesButtonClicked);
+
+    this.dialog
+      .querySelector("#not_now")
+      .addEventListener("click", this.closeDialogAndGoToCart);
+
+    this.dialog
+      .querySelector("#no")
+      .addEventListener("click", this.noButtonClicked);
+
+    this.dialog.querySelector("#bes_close").addEventListener("click", () => {
+      this.dialog.style.display = "none";
+    });
+
+    this.port.onMessage.addListener(this.applyConfig);
+    this.port.postMessage({ requestConfig: {} }); // TO DO: this must be at end of init
+  }
+
+  static applyConfig(msg) {
+    this.log.info("config recieved from backend" + JSON.stringify(msg.config));
+    this.config = msg.config;
+  }
+
+  static checkoutButtonClicked() {
+    if (
+      !this.config.albumOnCheckoutDisabled &&
+      !this.config.albumPurchasedDuringCheckout
+    ) {
+      this.dialog.style.display = "block";
+      return;
+    }
+
+    this.closeDialogAndGoToCart();
+  }
+
+  static closeDialogAndGoToCart() {
+    this.dialog.style.display = "none";
+
+    const clickEvent = new MouseEvent("click", {
+      view: window,
+      bubbles: true,
+      cancelable: false
+    });
+    let checkout_button = document.querySelector("#sidecartCheckout");
+    checkout_button.dispatchEvent(clickEvent);
+  }
+
+  static yesButtonClicked() {
+    let unit_price = this.dialog.querySelector("input").value;
+    this.log.info(`price ${unit_price}`);
+
+    let error_div = this.dialog.querySelector("#bes_checkout_error");
+    error_div.innerHTML = " ";
+    if (unit_price < 5) {
```

Comment:
I had this originally but thought it wasn't necessary since the "input" a "number" so entering other values is not possible.

before I changed it to a "number" input, it was a string input and Bandcamp's API will just try to add the item as 0 dollars and then complain that the item is not allowed for that pice